### PR TITLE
Feature/access controls

### LINKED ISSUE #74: Finalize Authentication and Authorization expectations/requirements
### Descriptive summary

Per the discussion with Kacy, we need to try and setup the following access control situation in this app.

~~**Try** to see if we can use IP ranges to allow anyone to _VIEW_ the `index` and `show` pages for recognitions. If this can't be done, we'll have to use SSO~~

1) Authenticated Library Staff users may view ALL recognitions EXCEPT those that are *suppressed*

2) Authenicated Library Staff may only _EDIT_ and _DELETE_ their *own* Recognitions

3) Administrators may _CRUD_ **all Recognitions**

### Rationale

Ensure the application is setup to be managed as expected by stakeholders.

### PR DESCRIPTION
**Pull request by mcritchlow**
Fixes #74 

#### Local Checklist
- [x] Tests written and passing locally?
- [x] Code style checked?
- [x] QA-ed locally?
- [x] Rebased with `master` branch?
- [x] Configuration updated (if needed)?

#### What does this PR do?
A few things:
- Introduces a landing page `front` that can be used for branding
- Add `show` and `index` to the `require_user` callback to enforce global auth for recognitions
- Add `authorize_editor` callback for recognitions to only allow Users to edit recognitions if they were the creator OR are an admin
- Store the requested URL in a session variable so the user can be directed there after login. Example a user follows a link to a recognition from the LiSN rss feed (recognition/123) and is then asked to authenticate. Previously they would have been redirected to the index page, which would have been very confusing. Now they should be routed to recognition/123 after authenticating
- Added tests for everything I could think of

##### Why are we doing this? Any context of related work?
References #74 

#### Where should a reviewer start?
Each commit is more or less a chunk of the work defined above. It may be worth working through those and their commit messages.

@VivianChu  - please review

### CODE REVIEW

File: spec/system/authorization_spec.rb [hunk 1]
```diff
@@ -1,6 +1,12 @@
 require 'rails_helper'
 
 RSpec.describe 'enforcing authorization in recognitions', type: :system do
+  before do
+    mock_valid_library_employee
+    omniauth_setup_shibboleth
+    mock_email_credential
```
Comment (VivianChu) [inline spec/system/authorization_spec.rb]: @mcritchlow - It's good to see that moving this mock higher in setup works.  Do you think if we should do the same thing for other tests to make it consistent?

File: app/helpers/auth_helper.rb [hunk 2]
```diff
@@ -18,8 +19,21 @@ def valid_administrator?
     User.administrator?(current_user.id) || current_user.developer?
   end
 
+  # Determines when the currently authenticate user is allowed to Edit/Update/Destroy a given recognition
+  # The recognition must belong to the current user or the current user must be an administrator
+  # @param [Recognition] recognition
+  def can_administrate?(recognition)
+    current_user.recognitions.include?(recognition) || User.administrator?(current_user.id)
```
Comment (VivianChu) [inline app/helpers/auth_helper.rb]: @mcritchlow -  I'm wondering how come you don't use the method valid_administrator? in this check for administrator.  Thanks.

**Comment by VivianChu** [comment]
@mcritchlow - 👍 overall.  I just have couple of questions above.

File: app/helpers/auth_helper.rb (hunk 2 above)
Comment (mcritchlow) [reply app/helpers/auth_helper.rb]: Ok this one is done.

File: spec/system/authorization_spec.rb (hunk 1 above)
Comment (VivianChu) [reply spec/system/authorization_spec.rb]: 6 spec files need it.  However, most of them relate to the email function.  I can update them and test them when this PR is merged and after the demo.

File: app/helpers/auth_helper.rb (hunk 2 above)
Comment (VivianChu) [reply app/helpers/auth_helper.rb]: Thanks Matt.  Looks good.

**Pull request merged**